### ISSUE TITLE
vue/html-quotes has bug for Quotation mark nesting

### ISSUE THREAD

**Issue by sept08** [vuejs/eslint-plugin-vue#337]
**Tell us about your environment**

* **ESLint Version:** 4.14.0
* **eslint-plugin-vue Version:** 4.0.0-beta.4
* **Node Version:** 8.9.1

**Please show your full configuration:**

```
'vue/html-quotes': [
      'error',
      'single'
],



```

**What did you do? Please include the actual source code causing the issue.**

```vue
<template>
  <div>
    <div :style="{height: tmpH + 'px'}">
  </div>
</template>




```

**What did you expect to happen?**
```vue
<div :style='{height: tmpH + "px"}'>
```
**What actually happened? Please include the actual, raw output from ESLint.**
```vue
<div :style='{height: tmpH + &quot;px&quot;}'>
```

**Comment by mysticatea** [comment]
Thank you for the report.

This is intentional behavior. `&quot;` means `'` in HTML. See [HTML entity](https://www.google.co.jp/search?q=html+entity).

**Comment by dougalg** [comment]
Actually, my mistake, it seems that Vue does parse this properly -- it's just a bit hard to read

**Comment by Apokly** [comment]
@dougalg Having same behaviour, looks a bit hard to read `&quot;` entities in source code. Do you have a workaround ?

**Comment by jackchoumine** [comment]
> **Tell us about your environment**
> 
> * **ESLint Version:** 4.14.0
> * **eslint-plugin-vue Version:** 4.0.0-beta.4
> * **Node Version:** 8.9.1
> 
> **Please show your full configuration:**
> 
> ```
> 'vue/html-quotes': [
>       'error',
>       'single'
> ],
> ```
> 
> **What did you do? Please include the actual source code causing the issue.**
> 
> ```
> <template>
>   <div>
>     <div :style="{height: tmpH + 'px'}">
>   </div>
> </template>
> ```
> 
> **What did you expect to happen?**
> 
> ```
> <div :style='{height: tmpH + "px"}'>
> ```
> 
> **What actually happened? Please include the actual, raw output from ESLint.**
> 
> ```
> <div :style='{height: tmpH + &quot;px&quot;}'>
> ```

Hello I found that use prettier formatting .vue file . single quotes become double  do you have this issue?
